bigchaindb/core.py

```diff
@@ -722,10 +660,14 @@ def get_last_voted_block(self):
                 r.table('bigchain', read_mode=self.read_mode)
                 .get(last_block_id))
 
-        return res
+        return Block.from_dict(res)
 
     def get_unvoted_blocks(self):
-        """Return all the blocks that has not been voted by this node."""
+        """Return all the blocks that have not been voted on by this node.
+
+        Returns:
+            block (list(dict)): a list of unvoted blocks
```

Comment:
> In any case, why does this return a list of dicts, when it could be returning a list of Blocks?

This method is also the reason why https://github.com/bigchaindb/bigchaindb/pull/641/files/76c796429055d180dcf3caaf2a95e1591e0d743c#r80649416 (`is_genesis_block`) can handle both dict-blocks as well as Block-blocks.

To understand why, you have to see where its used. Fun fact: It's only used once in the entire code base:

https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/pipelines/vote.py#L140

`initialize` runs only once when you freshly start bigchaindb. It basically simulates the changefeed and hence all blocks the newly initialized node has to work for. The changefeed is in json, so is this method.

My proposal (this should maybe handled outside of this PR) is to put this function into `vote.py`.